src/HealthCheck.Mvc/Controllers/MvcHealthCheckController.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.Remoting.Messaging;
+using System.Web.Mvc;
+using HealthCheck.Core;
+using HealthCheck.Mvc.Models;
+
+namespace HealthCheck.Mvc.Controllers
+{
+    public class MvcHealthCheckController : AsyncController
+    {
+        private const string Key = "result";
+        private readonly Core.HealthCheck _healthCheck;
+
+        public MvcHealthCheckController()
+        {
+            _healthCheck = new Core.HealthCheck(MvcHealthCheckСonfigurator.GetInstance().GetCheckers());
+        }
+
+        public void RunAsync()
+        {
+            AsyncManager.OutstandingOperations.Increment();
+            AsyncManager.Parameters[Key] = _healthCheck.Run().Result;
+            AsyncManager.OutstandingOperations.Decrement();
+        }
+
+        public ActionResult RunCompleted(HealthCheckResult result)
+        {            
+            var model = new HealthCheckResultModel(result);
```

Comment:
...also I would really prefer if you used the `HealthCheckResult` instance directly as view model instead of this new `HealthCheckResultModel` - that would make the "raw" output from the healthcheck endpoint look the same regardless of whether one is using the Nancy module or the MVC one. (You can still format it differently in the HTML view.)

Reply:
ok